app/hotels/src/navigation/singleHotel/SingleHotelStack.js

```diff
@@ -17,10 +17,40 @@ import GalleryGrid from '../../gallery/GalleryGrid';
 import { PaymentScreen } from '../../singleHotel/PaymentScreen';
 import AdditionalPropsInjector from './AdditionalPropsInjector';
 
-export default StackNavigator(
+const DetailStack = StackNavigator(
   {
     SingleHotel: {
-      screen: SingleHotelNavigationScreen,
+      screen: withMappedProps(SingleHotelNavigationScreen),
+      navigationOptions: {
+        headerLeft: null,
+        ...createTransparentHeaderStyle(RNDimensions.get('screen')),
+      },
+    },
+    GalleryGrid: {
+      screen: withMappedProps(GalleryGrid, AdditionalPropsInjector),
+    },
+    Payment: {
+      screen: withMappedProps(PaymentScreen),
+      navigationOptions: {
+        mode: 'modal',
+        headerTitle: (
```

Comment:
Can we move this to the PaymentScreen component?

Reply:
Probably yes

Reply:
But we will get this error 😄 

![image](https://user-images.githubusercontent.com/7705041/46679092-82ab9680-cbe6-11e8-986a-d1a1a99270be.png)

Reply:
@RobinCsl 🚀